src/components/ImageView/index.js

```diff
@@ -98,17 +94,22 @@ class ImageView extends PureComponent {
      * @param {SyntheticEvent} e
      */
     onContainerPress(e) {
-        if (this.state.isZoomed && !this.state.isDragging) {
+        let sX;
+        let sY;
+        if (this.isZoomed && !this.state.isDragging) {
             const {offsetX, offsetY} = e.nativeEvent;
-            const delta = this.getScrollOffset(offsetX, offsetY);
-            const sX = delta.offsetX;
-            const sY = delta.offsetY;
-            this.scrollableRef.scrollTop = sY * this.state.zoomScale;
-            this.scrollableRef.scrollLeft = sX * this.state.zoomScale;
+            const delta = this.getScrollOffset(offsetX / this.state.zoomScale, offsetY / this.state.zoomScale);
+            sX = delta.offsetX;
+            sY = delta.offsetY;
         }
 
-        if (this.state.isZoomed && this.state.isDragging && this.state.isMouseDown) {
+        if (this.isZoomed && this.state.isDragging && this.state.isMouseDown) {
```

Comment:
We need to use it state.isZoomed to detect the change in rendering. I set the class level because when the state changes, the information of the clicked point is incorrect on containerPress . We hold state change , make calculations then set isZoomed state. 
I know it looks bad. But if we let the state change, we were losing the clicked coordinates.